File: examples/02-reliability/go/retry_and_timeouts.go
package reliability

import (
	"time"

	"go.temporal.io/sdk/temporal"
	"go.temporal.io/sdk/workflow"
)

// Activity timeouts and retry policy live together on ActivityOptions, the same
// knobs as the Java ActivityOptions builder.
func ActivityOptionsExample(ctx workflow.Context) workflow.Context {
	return workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
		// One attempt must finish inside this window.
		StartToCloseTimeout: 5 * time.Minute,
		// The whole retry series must finish inside this window.
		ScheduleToCloseTimeout: 30 * time.Minute,
		RetryPolicy: &temporal.RetryPolicy{
			InitialInterval:    5 * time.Second,
			BackoffCoefficient: 2.0,
			MaximumInterval:    time.Minute,
			MaximumAttempts:    6,
		},
	})
}

File: examples/02-reliability/python/retry_and_timeouts.py
from datetime import timedelta

from temporalio import workflow
from temporalio.common import RetryPolicy

# `build_report` is a module-level @activity.defn function.


@workflow.defn
class ActivityOptionsExamples:
    @workflow.run
    async def run(self) -> None:
        await workflow.execute_activity(
            build_report,
            # One attempt must finish inside this window.
            start_to_close_timeout=timedelta(minutes=5),
            # The whole retry series must finish inside this window.
            schedule_to_close_timeout=timedelta(minutes=30),
            retry_policy=RetryPolicy(
                initial_interval=timedelta(seconds=5),
                backoff_coefficient=2.0,
                maximum_interval=timedelta(minutes=1),
                maximum_attempts=6,
            ),
        )
